lib/youtube/search/search.rb

```diff
@@ -0,0 +1,65 @@
+require "httparty"
+require "youtube/search/search_items"
+require "youtube/search/search_item"
+
+module YouTube
+  class Search
+    include HTTParty
+    base_uri 'https://www.googleapis.com'
+
+    attr_accessor :api_key
+
+    def initialize(api_key)
+      @api_key = api_key
+    end
+
+    # set pageToken value for the first page of search result
+    def first_page!
+      @page_token = ''
+    end
+
+    # set pageToken value for the next page
+    def next_page!
+      @page_token = @result["nextPageToken"] if @result
+    end
+
+    # set pageToken value for the previous page
+    def previous_page!
+      @page_token = @result["prevPageToken"]
+    end
+
+    def get_search_items(search_word)
+      SearchItems.new(search_response(search_word)).items
+    end
+
+    def first(pages = 1, search_word)
```

Comment:
Thank you for suggestion. But personally I don't think we are looking forward to change public methods right now. We can discuss further of course.